azurelinuxagent/common/protocol/wire.py

```diff
@@ -127,25 +130,6 @@ def get_ext_handler_pkgs(self, ext_handler):
     def get_goal_state(self):
         return self.client.get_goal_state()
 
-    def _download_ext_handler_pkg_through_host(self, uri, destination):
-        host = self.client.get_host_plugin()
-        uri, headers = host.get_artifact_request(uri, host.manifest_uri)
-        success = self.client.stream(uri, destination, headers=headers, use_proxy=False, max_retry=1)  # set max_retry to 1 because extension packages already have a retry loop (see ExtHandlerInstance.download())
-        return success
-
-    def download_ext_handler_pkg(self, uri, destination, headers=None, use_proxy=True):  # pylint: disable=W0613
-        direct_func = lambda: self.client.stream(uri, destination, headers=None, use_proxy=True, max_retry=1)
-        # NOTE: the host_func may be called after refreshing the goal state, be careful about any goal state data
-        # in the lambda.
-        host_func = lambda: self._download_ext_handler_pkg_through_host(uri, destination)
-
-        try:
-            success = self.client.send_request_using_appropriate_channel(direct_func, host_func) is not None
```

Comment:
This was a bug that would not detect errors downloading extensions. The stream() method used for the download use to return False on error and this is comparing against None. 

This is no longer an issue, since the refactored code now uses exceptions to report errors